src/spaces/nothing.jl

```diff
@@ -3,7 +3,7 @@ This file implements the "agent-space interaction API" for `nothing`, ie
 no space type
 =#
 
-function kill_agent!(agent::A, model::ABM{Nothing,A}) where {A<:AbstractAgent}
+function kill_agent!(agent::A, model::ABM{Nothing,A,Dict{Int,A}}) where {A<:AbstractAgent}
     delete!(model.agents, agent.id)
 end
```

Comment:
The functions `add_agent_to_model!` and `remove_agent_from_model!` dispatch on `::ABM` and `::UnkillableABM`.

Reply:
Shall I make a new PR to work on the internal API before continuing here? It seems that it would be very helpful to have these core functions as neat as possible before building on them.

Reply:
These functions are anyway best seen in the context of two or more variants of ABM, so it's fine for me to do this in this PR.

Reply:
In the meantime what's the proper way to implement addition and removal?